eng/pipelines/templates/jobs/live.tests.yml

```diff
@@ -201,32 +237,66 @@ jobs:
     # this step only makes sense when ctest has run
     condition: and(succeededOrFailed(), ne(variables['RunSamples'], '1'))
 
-    # Running Samples step.
-    # Will run samples described on a file name [service]-samples.txt within the build directory.
-    # For example keyvault-samples.txt.
-    # The file is written by CMake during configuration when building samples.
-  - bash: |
-      IFS=$'\n'
-      if [[ -f "./${{ parameters.ServiceDirectory }}-samples.txt" ]]; then
-        for sample in `cat ./${{ parameters.ServiceDirectory }}-samples.txt`
-        do
-          export AZURE_CLIENT_ID=$(${{parameters.ServiceDirectory}}_CLIENT_ID)
-          export AZURE_TENANT_ID=$(${{parameters.ServiceDirectory}}_TENANT_ID)
-          export AZURE_CLIENT_SECRET=$(${{parameters.ServiceDirectory}}_CLIENT_SECRET)
-          echo "**********Running sample: ${sample}"
-          bash -c "$sample"
-          status=$?
-          if [[ $status -eq 0 ]]; then
-            echo "*********Sample completed*********"
-          else
-            echo "*Sample returned a failed code: $status"
-            exit 1
-          fi
-        done
-      fi
-    workingDirectory: build
-    displayName: "Run Samples for : ${{ parameters.ServiceDirectory }}"
-    condition: and(succeeded(), eq(variables['RunSamples'], '1'))
+
+  - ${{ if parameters.UseFederatedAuth }}:
+      # Running Samples step.
+      # Will run samples described on a file name [service]-samples.txt within the build directory.
+      # For example keyvault-samples.txt.
+      # The file is written by CMake during configuration when building samples.
+    - bash: |
+        IFS=$'\n'
+        if [[ -f "./${{ parameters.ServiceDirectory }}-samples.txt" ]]; then
+          for sample in `cat ./${{ parameters.ServiceDirectory }}-samples.txt`
+          do
+            export AZURE_CLIENT_ID=$(${{parameters.ServiceDirectory}}_CLIENT_ID)
+            export AZURE_TENANT_ID=$(${{parameters.ServiceDirectory}}_TENANT_ID)
+            export AZURE_CLIENT_SECRET=$(${{parameters.ServiceDirectory}}_CLIENT_SECRET)
+            echo "**********Running sample: ${sample}"
+            bash -c "$sample"
+            status=$?
+            if [[ $status -eq 0 ]]; then
+              echo "*********Sample completed*********"
+            else
+              echo "*Sample returned a failed code: $status"
+              exit 1
+            fi
+          done
+        fi
+      workingDirectory: build
```

Comment:
Do we need `exit $LASTEXITCODE` here and in the else as well so the step fails if the pipeline if there is an error?

Reply:
Not needed in `bash` ... `exit $LASTEXITCODE` is required just for `AzurePowerShell@5`